py/embedded-server/deephaven_server/cli.py

```diff
@@ -61,4 +62,8 @@ def signal_handler(sig, frame):
         sys.exit(0)
 
     signal.signal(signal.SIGINT, signal_handler)
-    signal.pause()
\ No newline at end of file
+    try:
+        signal.pause()
+    except AttributeError:
+        # signal.pause() is not available on Windows
+        os.system("pause")
```

Comment:
It's not clear to me that this has the same semantics as `signal.pause()`. We should probably link to Windows "pause" documentation if we think it's correct to call.

Regardless, it's probably more appropriate to lookup `system.platform() == 'Windows'` instead of relying on an exceptional case?

Reply:
I did a little searching on this case using various AI tools.  Perplexity gave a solution that is worth consideration.  I haven't given it deep thought, but it looks like it might work and looks portable to all of the OSes.

-----

There is no direct equivalent of signal.pause() on Windows in Python. The signal.pause() function is Unix-specific and blocks the calling thread until a signal is received.
On Windows, you can achieve similar behavior using a queue and waiting on it indefinitely until an item is put into the queue from a signal handler. Here's an example:
```python
import signal
import threading

# Create a queue to block on
my_queue = threading.Event()

def signal_handler(sig, frame):
    # Trigger the queue when a signal is received
    my_queue.set()

# Set the signal handler
signal.signal(signal.CTRL_C_EVENT, signal_handler)

try:
    # Wait indefinitely until the queue is triggered
    my_queue.wait()
except KeyboardInterrupt:
    # Handle the interrupt gracefully
    print("Interrupted by user")
```
In this example, we create an Event object from the threading module, which acts as a queue. The signal_handler function is set to trigger the event when CTRL+C is pressed. The main thread then blocks indefinitely on my_queue.wait() until the signal handler sets the event, effectively pausing execution until a signal is received.
While not as elegant as signal.pause(), this approach allows you to pause execution and handle signals on Windows in a similar manner.

Reply:
Just saw this comment. I'll give that a shot

Reply:
Also seems like `input('')` works, which I just patched - any reason not to just use that?

Reply:
Could we use `c = input()` and catch `KeyboardInterruption` to make the behavior consistent on both Win and Nix?

Reply:
@jmao-denver I've done that and updated this review